fvm/storage_capacity_resolver.go

```diff
@@ -0,0 +1,64 @@
+package fvm
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+
+	"github.com/onflow/flow-go/fvm/state"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type StorageCapacityResolver func(state.Ledger, flow.Address) (uint64, error)
+
+func CreateResourceStorageCapacityResolver() StorageCapacityResolver {
+	return func(l state.Ledger, address flow.Address) (uint64, error) {
+		key := fmt.Sprintf("%s\x1F%s", "storage", "storageCapacity") // StorageCapacity resource key. Its the /storage/storageCapacity path
+		resource, err := l.Get(string(address.Bytes()), "", key)
+		if err != nil {
+			return 0, fmt.Errorf("could not load storage capacity resource: %w", err)
+		}
+
+		storedData, version := interpreter.StripMagic(resource)
+		commonAddress := common.BytesToAddress(address.Bytes())
+		storedValue, err := interpreter.DecodeValue(storedData, &commonAddress, []string{key}, version)
```

Comment:
Would be great if we could re-use `runtimeStorage.readValue` here.

`runtimeStorage` should maybe not take a whole `runtimeInterface`, but only a subset of it, i.e. we could create a sub-interface `runtime.StorageInterface` that has `GetValue`, `SetValue`, etc. and we could easily hook up the ledger.